### DISCUSSION TITLE
Should a field's type specify `null` when setting `nullable: true`?

### DISCUSSION THREAD

**Discussion by mmcgahan** [MichalLytek/type-graphql#842] [Q&A]
e.g. from the type-graphql docs:

```ts
  @Field({ nullable: true })
  averageRating?: number;
```

would it be more accurate/type-safe to include `null` in the field's type, i.e.

```ts
  @Field({ nullable: true })
  averageRating?: number | null;
```

I'm new to type-graphql but I think TS would be able to enforce better safety if the field type included `null`? If it doesn't matter, is there a simple explanation of why the GQL-related `nullable: true` is sufficient?

**Comment by MichalLytek** [comment]
Technically it should be `number | null` but thanks to the truthy/falsy mechanism in JS, people mostly do:
```ts
if (foo) {
  bar();
}
```
So for TS it makes no difference it it's only `undefined` or `undefined | null`.

Moreover, declaring field as `T | null` forces you to use `@Field(() => T)` which might be less convinent.

**Reply by gregoirechauvet** [reply]
I think it's not that reliable to do `if (foo)`. When `foo` is a boolean, you can't do this otherwise the `false` case will be handled as `undefined` and `null`. Same with `number` actually, `0` is falsy.
Especially, on nullable inputs, the client can omit the field (i.e. `undefined`), send `null`, or send a value. And to handle this we need to check both for `undefined` and `null`.

To help with this, is there a way to automatically convert `null` to `undefined`? Would that be a good idea? Or with a custom decorator maybe?

**Reply by MichalLytek** [reply]
you can do `if (foo != null)` which is checking for both `null` and `undefined`.

**Reply by gregoirechauvet** [reply]
That's true :+1: 
However, I still need to add `| null` to all the fields as `@mmcgahan` suggests, so I can make sure I handle all the cases properly.
